pynxtools_xps/phi/spe_pro_phi.py

```diff
@@ -0,0 +1,1549 @@
+                "profiling_depth_recalibration",
+                "profiling_sputter_mode",
+                "depth_calibration_definition",
+                "profiling_no_depth_regions",
+            ],
+            # "unused": [
+            # "peak_to_noise_ratio_state",
+            # "platform",
+            # "platen_id",
+            # "presputter",
+            # "register_image",
+            # "register_image_interval",
+            # "register_image_last",
+            # "register_image_mode",
+            # ],
+        }
+
+        for spectrum in spectra:
+            self._update_xps_dict_with_spectrum(spectrum, key_map)
+
+    def _update_xps_dict_with_spectrum(
+        self, spectrum: Dict[str, Any], key_map: Dict[str, List[str]]
+    ):
+        """
+        Map one spectrum from raw data to NXmpes-ready dict.
+
+        """
+        # pylint: disable=too-many-locals,duplicate-code
+        try:
+            group_parent = f'{self._root_path}/Group_{spectrum["group_name"]}'
+            region_parent = f'{group_parent}/Region_{spectrum["spectrum_type"]}'
+        except KeyError:
+            region_parent = f'{self._root_path}/Region_{spectrum["spectrum_type"]}'
+        file_parent = f"{region_parent}/file_info"
+        instrument_parent = f"{region_parent}/instrument"
+        analyser_parent = f"{instrument_parent}/analyser"
+
+        path_map: Dict[str, str] = {
+            "file_info": f"{file_parent}",
+            "user": f"{region_parent}/user",
+            "instrument": f"{instrument_parent}",
+            "xray_source": f"{instrument_parent}/xray_source",
+            "beam": f"{instrument_parent}/beam",
+            "analyser": f"{analyser_parent}",
+            "collectioncolumn": f"{analyser_parent}/collectioncolumn",
+            "energydispersion": f"{analyser_parent}/energydispersion",
+            "detector": f"{analyser_parent}/detector",
+            "manipulator": f"{instrument_parent}/manipulator",
+            "c60_ion_gun": f"{instrument_parent}/c60_ion_gun",
+            "defect_positioner": f"{instrument_parent}/defect_positioner",
+            "flood_gun": f"{instrument_parent}/flood_gun",
+            "gcib": f"{instrument_parent}/gcib",
+            "neutral_ion_gun": f"{instrument_parent}/neutral_ion_gun",
+            "sputter_gun": f"{instrument_parent}/sputter_gun",
+            "process": f"{region_parent}/process",
+            "sample": f"{region_parent}/sample",
+            "data": f"{region_parent}/data",
+            "region": f"{region_parent}/region",
+            "profiling": f"{region_parent}/profiling",
+        }
+
+        for grouping, spectrum_keys in key_map.items():
+            root = path_map[str(grouping)]
+            for spectrum_key in spectrum_keys:
+                try:
+                    mpes_key = spectrum_key
+                    self._xps_dict[f"{root}/{mpes_key}"] = spectrum[spectrum_key]
+                except KeyError:
+                    pass
```

Comment:
Maybe a bit cleaner and w/o slow try...except + it's clearer what is actually checked:
```suggestion
                mpes_key = spectrum_key
                self._xps_dict[f"{root}/{mpes_key}"] = spectrum.get(spectrum_key)

```

that way the key would be filled with `None` if the key does not exists. If this is not fine maybe like this
```suggestion
                mpes_key = spectrum_key
                if spectrum_key in spectrum:
                    self._xps_dict[f"{root}/{mpes_key}"] = spectrum[spectrum_key]

```

Reply:
I went with the first suggestion, that works.